gluten-core/src/main/scala/io/glutenproject/expression/AggregateFunctionsBuilder.scala

```diff
@@ -27,19 +27,30 @@ object AggregateFunctionsBuilder {
   def create(args: java.lang.Object, aggregateFunc: AggregateFunction): Long = {
     val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
 
-    var substraitAggFuncName = getSubstraitFunctionName(aggregateFunc)
+    // First handle the custom aggregate functions
+    val (substraitAggFuncName, inputTypes) =
+      if (
+        ExpressionMappings.expressionExtensionTransformer.extensionExpressionsMapping.contains(
+          aggregateFunc.getClass)
+      ) {
+        ExpressionMappings.expressionExtensionTransformer.buildCustomAggregateFunction(
+          aggregateFunc)
+      } else {
+        val substraitAggFuncName = getSubstraitFunctionName(aggregateFunc)
 
-    // Check whether each backend supports this aggregate function.
-    if (
-      !BackendsApiManager.getValidatorApiInstance.doExprValidate(
-        substraitAggFuncName.get,
-        aggregateFunc)
-    ) {
-      throw new UnsupportedOperationException(
-        s"Aggregate function not supported for $aggregateFunc.")
-    }
+        // Check whether each backend supports this aggregate function.
+        if (
+          !BackendsApiManager.getValidatorApiInstance.doExprValidate(
+            substraitAggFuncName.get,
+            aggregateFunc)
+        ) {
+          throw new UnsupportedOperationException(
+            s"Aggregate function not supported for $aggregateFunc.")
+        }
 
-    val inputTypes: Seq[DataType] = aggregateFunc.children.map(child => child.dataType)
+        val inputTypes: Seq[DataType] = aggregateFunc.children.map(child => child.dataType)
```

Comment:
The `inputTypes` looks as same as that produced by `buildCustomAggregateFunction`. Maybe, we can move it outside the if/else block.

Reply:
In some cases, the `inputTypes` will be obtained using the other logic, it's better to support using custom logic to get `inputTypes`